src/shogun/machine/GLM.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Tej Sukhatme
+ */
+
+#include <shogun/features/DotFeatures.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/lib/observers/ObservedValueTemplated.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/machine/LinearMachine.h>
+#include <shogun/machine/GLM.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/RandomNamespace.h>
+#include <shogun/optimization/ElasticNetPenalty.h>
+#include <shogun/optimization/SGDMinimizer.h>
+#include <shogun/optimization/GradientDescendUpdater.h>
+#include <shogun/optimization/ConstLearningRate.h>
+
+#include<iostream>
+// #include <utility>
+#include <cmath>
+
+using namespace shogun;
+
+
+
+
+
+GLM::GLM()
+{
+	std::cout<<"Entered non-parameterized constructor.\n";
+	SG_ADD(&distribution, "distribution_type", "variable to store name of distribution type", ParameterProperties::HYPER);
+	SG_ADD(&m_eta, "eta", "threshold parameter that linearizes the exp() function above eta", ParameterProperties::HYPER);
+	SG_ADD(&m_lambda, "lambda", "regularization parameter of penalty term", ParameterProperties::HYPER);
+	SG_ADD(&m_alpha, "alpha", "weighting between L1 penalty and L2 penalty term of the loss function", ParameterProperties::HYPER);
+	SG_ADD(&m_tolerance, "tolerance", "convergence threshold or stopping criteria", ParameterProperties::HYPER);
+	
+	m_gradient_updater = std::make_shared<GradientDescendUpdater>();
+	m_learning_rate = std::make_shared<ConstLearningRate>();
+	m_penalty = std::make_shared<ElasticNetPenalty>();
+	m_cost_function = std::make_shared<GLMCostFunction>();
+
+}
+
+GLM::GLM(GLM_DISTRIBUTION distr, float64_t alpha, float64_t lambda, float64_t learning_rate, int32_t max_iterations, float64_t tolerance, float64_t eta): RandomMixin<IterativeMachine<LinearMachine>>()
```

Comment:
Here you should also call the default constructor. If you don't do that, the various `m_gradient_updater`, `m_learning_rate` won't be initialized. 

```c++
GLM::GLM(GLM_DISTRIBUTION distr, float64_t alpha, float64_t lambda, float64_t learning_rate, int32_t max_iterations, float64_t tolerance, float64_t eta): GLM()

```

Reply:
This should fix your segfaults.

Reply:
Yes, that was silly of me, sorry.